kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h

```diff
@@ -1224,10 +1224,11 @@ class ResidualBasedBlockBuilderAndSolver
     {
         Parameters default_parameters = Parameters(R"(
         {
-            "name"                               : "block_builder_and_solver",
-            "block_builder"                      : true,
-            "diagonal_values_for_dirichlet_dofs" : "use_max_diagonal",
-            "silent_warnings"                    : false
+            "name"                                 : "block_builder_and_solver",
+            "block_builder"                        : true,
+            "diagonal_values_for_dirichlet_dofs"   : "use_max_diagonal",
+            "silent_warnings"                      : false,
+            "advanced_builder_and_solver_settings" : {}

```

Comment:
```suggestion
```

not necessary due to #7290

Reply:
The parameters must be at least exist, or the validation will crash

Reply:
I dont understand, can you explain why/where?

Reply:
When you use the method ValidateAndAssignSettings you need to have all the parameters from the parameters to be checked defined in the base class, or will crash when passing throw the base class. As it is not  RecursivelyValidateAndAssignSettings, just the name will be enough, that's why I defined the parameters as "advanced_builder_and_solver_settings", so can be adapted to any derived classes

Reply:
but the derived classes should add the additional parameters themselves so this doesn't happen. This is exactly what #7290 does

Reply:
but this is the Strategy not the B&S

Reply:
It is exactly the same concept, this is due to the Parameters behavior, not from the B&S

Reply:
If I removed, it simply won't work (that's why it stopped compiling after I solved the merge conflicts this weekend)

Reply:
Ok then I will try myself

Reply:
compiling...

Reply:
now should work